src/components/Login.js

```diff
@@ -0,0 +1,92 @@
+import React, { Component } from 'react';
+import { AUTH_TOKEN } from '../constants';
+import { Mutation } from 'react-apollo';
+import gql from 'graphql-tag';
+
+const SIGNUP_MUTATION = gql`
+  mutation SignupMutation($email: String!, $password: String!, $name: String!) {
+    signup(email: $email, password: $password, name: $name) {
+      token
+    }
+  }
+`;
+
+const LOGIN_MUTATION = gql`
+  mutation LoginMutation($email: String!, $password: String!) {
+    login(email: $email, password: $password) {
+      token
+    }
+  }
+`;
+
+class Login extends Component {
+  state = {
+    login: true, // switch between Login and SignUp
+    email: '',
+    password: '',
+    name: ''
+  };
+
+  render() {
+    const { login, email, password, name } = this.state;
+    return (
+      <div>
+        <h4 className="mv3">{login ? 'Login' : 'Sign Up'}</h4>
+        <div className="flex flex-column">
+          {!login && (
+            <input
+              value={name}
+              onChange={e => this.setState({ name: e.target.value })}
+              type="text"
+              placeholder="Your name"
+            />
+          )}
+          <input
+            value={email}
+            onChange={e => this.setState({ email: e.target.value })}
+            type="text"
+            placeholder="Your email address"
+          />
+          <input
+            value={password}
+            onChange={e => this.setState({ password: e.target.value })}
+            type="password"
+            placeholder="Password"
+          />
+        </div>
+        <div className="flex mt3">
+          <Mutation
+            mutation={login ? LOGIN_MUTATION : SIGNUP_MUTATION}
+            variables={{ email, password, name }}
+            onCompleted={data => this._confirm(data)}
+          >
+            {mutation => (
+              <div className="pointer mr2 button" onClick={mutation}>
```

Comment:
Same here, missing error handling + validation, would be nice if you could implement something 🙃 

Also, bit of feedback: should be using `button` and `form` elements here

Reply:
Super! I will try to make this a bit nicer with some user feedback 😎